semantic_conventions/server.yaml

```
@@ -23,6 +23,8 @@ groups:
         brief: The domain name of an immediate peer.
         examples: ['proxy.example.com']
         note: Typically observed from the client side, and represents a proxy or other intermediary domain name.
+        requirement_level:
+          recommended: If different than `server.address`.
       - id: socket.address
         type: string
         brief: Physical server IP address or Unix socket address.
```

Not directly related, but I wonder if we should clarify:

```suggestion
        brief: Physical server IP address or Unix socket address. If set from the client, should simply use the socket's peer address, and not attempt to find any actual server IP (i.e., if set from client, this may represent some proxy server instead of the logical server).
```